src/main/java/lotto/controller/LottoController.java

```diff
@@ -0,0 +1,92 @@
+package lotto.controller;
+
+import static lotto.config.LottoConfig.PURCHASE_DIVISIBLE_AMOUNT;
+
+import java.util.HashSet;
+import java.util.List;
+import lotto.config.LottoServices;
+import lotto.config.LottoValidators;
+import lotto.domain.PurchasedLotto;
+import lotto.domain.WinningLotto;
+import lotto.dto.LottoResultDto;
+import lotto.dto.WinningLottoDto;
+import lotto.validator.BonusNumberValidator;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+public class LottoController{
+    private final LottoServices services;
+    private final LottoValidators validators;
+    private int purchaseAmount;
+    private int dividedPurchaseAmount;
+    private HashSet<PurchasedLotto> purchasedLotto;
+    private List<Integer> winningNumbers;
+    private int bonusNumber;
+    private LottoResultDto lottoResultDto;
+
+    public LottoController(LottoServices services, LottoValidators validators) {
+        this.services = services;
+        this.validators = validators;
+    }
+
+    public void run() {
+        setPurchaseAmount();
+        printPurchasedLottoAmount();
+        purchaseLotto();
+        setWinningNumbers();
+        setBonusNumber();
+        compare();
+        printLottoResult();
+    }
+
+    private void setPurchaseAmount() {
+        purchaseAmount = services.lottoSettingService.selectPurchaseAmount(validators.purchaseAmountValidator, InputView::getUserInput,
+                OutputView::printPurchaseAmountRequestMessage,  OutputView::printErrorMessage);
+    }
```

Comment:
인자 각각의 길이가 길어보이는데 분리해서 작성하는 방법은 어떨까요?
```suggestion
        purchaseAmount = services.lottoSettingService.selectPurchaseAmount(
                validators.purchaseAmountValidator, 
                InputView::getUserInput,
                OutputView::printPurchaseAmountRequestMessage,  
                OutputView::printErrorMessage);
```